app/Jobs/FetchCollectionVolume.php

```diff
@@ -44,9 +45,26 @@ public function handle(): void
             contractAddress: $this->collection->address
         );
 
-        $this->collection->update([
-            'volume' => $volume,
-        ]);
+        DB::transaction(function () use ($volume) {
+            if ($volume !== null) {
+                $this->collection->volumes()->create([
+                    'volume' => $volume,
+                ]);
+
+                $this->collection->update([
+                    'volume' => $volume,
+                    'avg_volume_24h' => $this->collection->averageVolumeSince(now()->subDays(1)),
+                    'avg_volume_7d' => $this->collection->averageVolumeSince(now()->subDays(7)),
+                    'avg_volume_1m' => $this->collection->averageVolumeSince(now()->subMonths(1)),
```

Comment:
```suggestion
                    'avg_volume_24h' => $this->collection->averageVolumeSince(now()->subDays(1)),
                    'avg_volume_7d' => $this->collection->averageVolumeSince(now()->subDays(7)),
                    'avg_volume_30d' => $this->collection->averageVolumeSince(now()->subDays(30)),
```

keeps it aligned with the days used